### PR TITLE
Final Bot Shooter Pivot

### PR DESCRIPTION
**Pull request by ChaoticChaosTheGreat**
yessss i did it

idasilhdsjkhfs

### CODE REVIEW

**Comment by ChaoticChaosTheGreat** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Team3256/FRC_Programming_2024/207?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#207** <a href="https://app.graphite.dev/github/pr/Team3256/FRC_Programming_2024/207?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#206** <a href="https://app.graphite.dev/github/pr/Team3256/FRC_Programming_2024/206?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ChaoticChaosTheGreat and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ChaoticChaosTheGreat** [comment]
Maybe make this the new final bot branch ?

File: src/main/java/frc/robot/subsystems/pivotshooter/PivotShooter.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    pivotMotorConfigs.CurrentLimits.StatorCurrentLimitEnable = enableStatorLimit;
+    pivotMotorConfigs.CurrentLimits.StatorCurrentLimit = statorLimit;
+    pivotMotor.getConfigurator().apply(pivotMotorConfigs);
+  }
+
+  private void configureSimHardware() {
+    configureRealHardware();
+    pivotMotorSim = pivotMotor.getSimState();
+    pivotMotorSim.setSupplyVoltage(12);
+    pivotModel =
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/pivotshooter/PivotShooter.java:125]: insane

File: src/main/java/frc/robot/subsystems/pivotshooter/commands/PivotShooterAutoAimSidEdition.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    } else if (ally.isPresent()) {
+      speakerId = Constants.PivotingShooter.kSpeakerAprilTagBlue;
+    } else {
+      speakerId = Constants.PivotingShooter.kSpeakerAprilTagRed;
+    }
+  }
+
+  @Override
+  public void execute() {
+    if (Limelight.getFiducialID("limelight") == speakerId) {
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/pivotshooter/commands/PivotShooterAutoAimSidEdition.java:41]: this is the exact same code as the previous autoaim, except that the distance is fetched from $\sqrt{\text{tx}^2 + \text{ty}^2}$. Perhaps feature flag?

File: src/main/java/frc/robot/subsystems/pivotshooter/PivotShooter.java (hunk 1 above)
Comment (ThatXliner) [reply src/main/java/frc/robot/subsystems/pivotshooter/PivotShooter.java:125]: as in "yay you did sim as well"

File: src/main/java/frc/robot/subsystems/pivotshooter/commands/PivotShooterAutoAimSidEdition.java (hunk 2 above)
Comment (ChaoticChaosTheGreat) [reply src/main/java/frc/robot/subsystems/pivotshooter/commands/PivotShooterAutoAimSidEdition.java:41]: sid requested :(

**Pull request closed without merging**